### ISSUE TITLE
Should we even have profiles

### ISSUE THREAD

**Issue by fgmccabe** [WebAssembly/profiles#2]
Allowing different flavors of what engines can support risks long term fracturing of the wasm ecosystem.

In addition, from the pov of an engine provider, they will be under long term pressure to keep up to date with the whole of the specification. That is in their interest because it enhances the probability that their engine will be useful.

From the pov of a wasm module developer, they will be motivated to go for the greatest common denominator: i.e., they will generate code that runs on as many engines as possible.

However, there is a similar but different issue that we **should** address - the consequences of having Wasm be a 'living document'. That implies that any given engine will likely be behind the current standard to varying extents.

**Comment by titzer** [comment]
I agree with Andreas in that profiles are important to define a blessed
subset of Wasm that is suitable for a domain. For example, embedded systems
that are based on CPUs without SIMD support would benefit from a profile
definition that excludes it.

However I also agree with Francis about the "living document" problem. I
think we need a more permanent way to nail down a specification version
number. It might be as simple as tagging it with a year.

**Comment by rossberg** [comment]
#3 has more discussion of this question.
